### PR TITLE
Deprecated `use_auth_token`

### LINKED ISSUE #1836: Pushing to the hub with a token
### System Info

```shell
optimum main
```

### Who can help?

@echarlaix @fxmarty 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

The current integration with the hub doesn't allow pushing to it with some tokens because it requires the user's `email` and `fullname` (not present with fine-grained tokens).
```python
user = api.whoami(huggingface_token)
self.git_config_username_and_email(git_email=user["email"], git_user=user["fullname"])
```
to reproduce, use a fine-grained token to push to a repo (that the token has write access to) or simply `whoami(token="fine-grained-token-you-created")`, the output won't contain the `email` field.

I'm deprecating `use_auth_token` in a PR and will fix this on my way.

### Expected behavior

to push to hub

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

This PR deprecates the use `use_auth_token` in favor of `token`.

Fixes #1836 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

@fxmarty @echarlaix @mht-sharma

### CODE REVIEW

File: optimum/configuration_utils.py [hunk 1]
```diff
@@ -205,6 +218,15 @@ def _get_config_dict(
         from_auto_class = kwargs.pop("_from_auto", False)
         commit_hash = kwargs.pop("_commit_hash", None)
 
+        if use_auth_token is not None:
+            warnings.warn(
```
Comment (fxmarty) [inline optimum/configuration_utils.py:222]:
```suggestion
            logger.warning(
```

Comment (fxmarty) [reply optimum/configuration_utils.py:222]: Interestingly Transformers also uses warnings.warn, not sure why the mix?

File: optimum/exporters/tasks.py [hunk 2]
```diff
@@ -18,12 +18,14 @@
...
 import itertools
 import os
+import warnings
 from functools import partial
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Set, Tuple, Type, Union
 
 import huggingface_hub
 from huggingface_hub.constants import HUGGINGFACE_HUB_CACHE
+from huggingface_hub.utils import OfflineModeIsEnabled
```
Comment (fxmarty) [inline optimum/exporters/tasks.py:28]: Good catch, thank you for fixing. Maybe even move to `huggingface_hub.errors.OfflineModeIsEnabled`? https://github.com/huggingface/huggingface_hub/blob/v0.23.3/src/huggingface_hub/errors.py#L16

As I guess the `huggingface_hub.utils.OfflineModeIsEnabled` path is deprecated?

**Review by fxmarty — APPROVED** [review]
LGTM

File: optimum/configuration_utils.py (hunk 1 above)
Comment (IlyasMoutawwakil) [reply optimum/configuration_utils.py:222]: my guess is that this is specific for FutureWarning, should we use logger everywhere ?

Comment (IlyasMoutawwakil) [reply optimum/configuration_utils.py:222]: I asked in https://huggingface.slack.com/archives/C01N44FJDHT/p1719313821448529
the answer is in https://stackoverflow.com/a/14762106 which refers to `logging` documentation

> warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted

**Pull request merged**